sdk/storage/azure-storage-file-datalake/tests/test_file_system.py

```diff
@@ -668,104 +670,114 @@ def test_undelete_file_with_version_id(self, datalake_storage_account_name, data
-    #     )
-    #     assert len(response) == 5
-    #     assert response[0].status_code == 202
-    #     assert response[1].status_code == 202
-    #     assert response[2].status_code == 202
-    #     assert response[3].status_code == 409
-    #     assert response[4].status_code == 404
+    @DataLakePreparer()
+    def test_delete_files_simple_no_raise(self, datalake_storage_account_name, datalake_storage_account_key):
+        # Arrange
+        self._setUp(datalake_storage_account_name, datalake_storage_account_key)
+        filesystem = self._create_file_system("fs1")
+        data = b'hello world'
+
+        try:
+            # create file1
+            filesystem.get_file_client('file1').upload_data(data, overwrite=True)
+
+            # create file2
+            file2 = filesystem.get_file_client('file2')
+            file2.upload_data(data, overwrite=True)
+
+            # create file3
+            file3 = filesystem.get_file_client('file3')
+            file3.upload_data(data, overwrite=True)
+
+            # create dir1
+            # empty directory can be deleted using delete_files
+            filesystem.get_directory_client('dir1').create_directory(),
+
+            # create dir2
+            dir2 = filesystem.get_directory_client('dir2')
+            dir2.create_directory()
+
+        except:
+            pass
+
+        # Act
+        response = filesystem.delete_files(
+            'file1',
+            'file2',
+            'file3',
+            'dir1',
+            'dir2',
+            raise_on_any_failure=False
+        )
+
+        # Assert
+        self.assertEqual(len(response), 0)
+
+    @DataLakePreparer()
+    def test_delete_files_with_failed_subrequest(self, datalake_storage_account_name, datalake_storage_account_key):
+        # Arrange
+        self._setUp(datalake_storage_account_name, datalake_storage_account_key)
+        filesystem = self._create_file_system("fs2")
+        data = b'hello world'
+
+        try:
+            # create file1
+            filesystem.get_file_client('file1').upload_data(data, overwrite=True)
+
+            # create file2
+            file2 = filesystem.get_file_client('file2')
+            file2.upload_data(data, overwrite=True)
+
+            # create file3
+            file3 = filesystem.get_file_client('file3')
+            file3.upload_data(data, overwrite=True)
+
+            # create dir1
+            dir1 = filesystem.get_directory_client('dir1')
+            dir1.create_file("file4")
+
+            # create dir2
+            dir2 = filesystem.get_directory_client('dir2')
+            dir2.create_directory()
+
+        except:
+            pass
+
+        # Act
+        response = filesystem.delete_files(
+            'file1',
+            'file2',
+            'file3',
+            'dir1',  # dir1 is not empty
+            'dir8',  # dir8 doesn't exist
+            raise_on_any_failure=False
+        )
+        tuple1 = response[0]
```

Comment:
nit: I think having these accessors before line 759 (asserting the length) sort of defeats the purpose of that assert? If the length is less, these may throw before that assert. I would remove these and just do 2D accessors in all the assert statements (`tuple1[0][0]`). Or you could just move these below the asset on 759.

Reply:
Ah good point! Will change to 2D accessors 😄